src/psyclone/psyir/nodes/assignment.py

```diff
@@ -188,16 +190,16 @@ def reference_accesses(self, var_accesses):
     @property
     def is_array_range(self):
         '''
-        returns: True if the lhs of the assignment is an array with at \
-            least one of its dimensions being a range and False \
-            otherwise.
+        returns: True if the lhs of the assignment is an array access with at \
+            least one of its dimensions being a range and False otherwise.
         rtype: bool
 
         '''
-        from psyclone.psyir.nodes import ArrayReference, Range
-        if not isinstance(self.lhs, ArrayReference):
-            return False
-        return any(dim for dim in self.lhs.children if isinstance(dim, Range))
+        if isinstance(self.lhs, (ArrayReference, StructureReference)):
```

Comment:
I expect I am worrying unduly but this test is a little looser than the previous one, in that it considers that any `Range` node within an `ArrayReference` or `StructureReference` implies that this access is an array range. The particular example I am thinking about is an array slice within an expression. For example `a(min(b(:))) =`

Reply:
No, that's a fair point. I had a quick think about it (when I did it) and couldn't think of an example where it would break but you've found one! I'll improve it.

Reply:
In this situation (and maybe others) I think it would be helpful if we had the concept of operations that perform reductions, or alternatively, whether they return scalars or arrays. Maybe that comes back to the issue of type-checking PSyIR expressions?

Reply:
For completeness, your example should actually use `minval` which we don't yet support in the PSyIR. However, we do support `SUM` so that's what I've used in the test.

Reply:
Thanks.